src/Language/Haskell/Ghcid.hs

```diff
@@ -77,8 +83,13 @@ startGhci cmd directory echo = do
                     Just msg -> return msg
     r <- parseLoad <$> f ""
     writeIORef echo False
-    return (Ghci f,r)
 
+    let ghci = Ghci (ph, f)
+#ifndef mingw32_HOST_OS
+    tid <- myThreadId
+    installHandler keyboardSignal (Catch (interrupt ghci (Just "") True >> stopGhci ghci >> throwTo tid ExitSuccess)) Nothing
+#endif
```

Comment:
>What is the installHandler doing?

This takes care of ctrl-c not terminating my spawned test process(which can run forever) when exiting ghcid.  If there is another way, let me know :)

>As far as I can tell, if you press a key in the GUI, it interrupts the running ghci process then kills the entire process. Is this an orthogonal feature?

`keyboardSignal` is actually an alias of `sigINT`.  I know! misleading name, right, but probably has a historical meaning? :joy: 

> Why is it calling interrupt, given exiting the parent process kills the child process anyway? Why both interrupt and stopGhci?

This should be only interrupt + trowTo. 
`stopGhci` is a leftover from different approaches I was trying. By the way, stopGhci without trowTo left the process hanging